src/data/move.ts

```diff
@@ -1420,6 +1420,11 @@ export class RecoilAttr extends MoveEffectAttr {
       return false;
     }
 
+    // Chloroblast and Struggle should not deal recoil damage if the move was not successful
+    if (this.useHp && [ MoveResult.FAIL, MoveResult.MISS ].includes(user.getLastXMoves(1)[0].result)) {
```

Comment:
```suggestion
    if (this.useHp && [ MoveResult.FAIL, MoveResult.MISS ].includes(user.getLastXMoves(1)[0]?.result)) {
```
Just in case (similarly to #4665)